### DISCUSSION TITLE
Error building from source: Could not find one of 'package.json' manifest files in the package

### DISCUSSION THREAD

**Discussion by phodara** [mriscoc/Ender3V2S1#379] [General]
When I try building this version I keep getting an error-
Processing STM32F103RE_creality (board: genericSTM32F103RE; platform: ststm32@~12.1; framework: arduino)
-----------------------------------------------------------------------------Error: Could not find one of 'package.json' manifest files in the package 
I have cloned the repository correctly, synced the repository, created a new branch and I have Copied the correct configuration over for the Ender3 V2 board 4.22 with BLT.  Any idea what could be causing the error?
Thank you.

**Comment by mriscoc** [comment]
Error in PlatformIO installation.
https://community.platformio.org/t/upload-error-could-not-find-one-of-package-json-manifest-files-in-the-package/28149/7

**Reply by phodara** [reply]
I deleted the whole file structure numerous time and did git clone from the source, also did a resync. I also tried creating a unique local branch.
I may be doing something stupid but here are
my steps:
Work environment Windows10
1). git clone the repository to ender3v2\
2.) In github desktop I sync the repository
3).  I copy all the files from ender3v2\configurations\Ender3V2-422-BLT  and place them in  \ender3v2\Marlin
4). Then in VS Code I open the folder ender3v2\
5).  Launch auto build marlin, show ABM panel 
6). CLick build on STM32F103RE_creality (512k)

And then boom:
```
Processing STM32F103RE_creality (board: genericSTM32F103RE; platform: ststm32@~12.1; framework: arduino)
-------------------------------------------------------------------------------Error: Could not find one of 'package.json' manifest files in the package      
PS C:\Users\PAULXXXXX\Desktop\3DPrinting\EnderPrinter\Firmware\ender3v2> echo "done" >"C:\Users\PAULXX~1\AppData\Local\Temp\ipc"
```
Any assistance would be greatly appreciated

**Reply by phodara** [reply]
Not sure why github drew lines through the above text.  There are no lines in the terminal output.

**Reply by mriscoc** [reply]
The problem is not the repository or sources, it is your PlatformIO installation in VSCode.

**Reply by phodara** [reply]
Ok I will look into that today.  Thank you.

**Comment by phodara** [comment]
I made some progress today.  I went ahead and reinstalled windows git, platform IO and Marlin autobuild.  As I mentioned before I copied the files from Ender3V2-422-BLT including the platformio.ini file to the Marlin Directory.  As an experiment I brought up the ABM panel and tried to to build all the environments.  My understanding is the correct environment for my Ender 3V2 4.2.2 BLT should be STM32F103RE_creality(512k) all builds failed except for STM32F103RE_creality_maple (512K).  Was I building the wrong environment what's the difference between the 'STM..._creality' and the 'STM..._creality_maple'?   Thank you again for your time.

**Reply by mriscoc** [reply]
Maple was deprecated, use `STM32F103RE_creality` if you have a RET6 SoC or `STM32F103RC_creality` for RCT6.

**Reply by DARKFiB3R** [reply]
What is the _xfer environment for?

**Reply by mriscoc** [reply]
> What is the _xfer environment for?

For direct transfer the firmware using vscode in a test environment.

**Reply by DARKFiB3R** [reply]
Thank you.

**Comment by phodara** [comment]
Now I am really confused,  This link says this firmware should run with both the RET6 and the RCT6 see: https://github.com/mriscoc/Ender3V2S1/discussions/105.  Also if I switch from _maple to _creality do I have to download and install additional libraries and is this the issue with platformio?  The last firmware firmware I compiled without had no issues.  Is there a step by step guide with prerequisites for this version?  I am sorry about my ignorance regarding this install.

**Reply by phodara** [reply]
Fixed!!! Not sure what happened I uninstalled VS Code completely deleted all registry entries, I removed entire platformio dir structure, then reinstalled everything and it works.  You are probably correct I think it was an issue with the platformio configuration in VSCode.  I would have preferred to trouble shoot and figure out exactly what was wrong, but at least it works now.  Thank you.
